src/palace/manager/api/odl/api.py

```diff
@@ -422,26 +339,59 @@ def _checkout(
             raise NoAvailableCopies()
         loan, ignore = license.loan_to(patron)
 
+        identifier = loan.license.identifier
+        checkout_id = str(uuid.uuid4())
+        if self.collection is None:
+            raise PalaceValueError(f"Collection not found: {self.collection_id}")
+        default_loan_period = self.collection.default_loan_period(loan.patron.library)
+
+        requested_expiry = utc_now() + datetime.timedelta(days=default_loan_period)
+        patron_id = patron.identifier_to_remote_service(licensepool.data_source)
+        library_short_name = loan.patron.library.short_name
+
+        db = Session.object_session(loan)
+        patron = loan.patron
+        hasher = self._get_hasher()
+
+        unhashed_pass = self._credential_factory.get_patron_passphrase(db, patron)
+        hashed_pass = unhashed_pass.hash(hasher)
+        self._credential_factory.set_hashed_passphrase(db, patron, hashed_pass)
+        encoded_pass = base64.b64encode(binascii.unhexlify(hashed_pass.hashed))
+
+        notification_url = self._url_for(
+            "odl_notify",
+            library_short_name=library_short_name,
+            loan_id=loan.id,
+            _external=True,
+        )
+
+        # We should never be able to get here if the license doesn't have a checkout_url, but
+        # we assert it anyway, to be sure we fail fast if it happens.
+        assert license.checkout_url is not None
+        url_template = URITemplate(license.checkout_url)
+        checkout_url = url_template.expand(
+            id=str(identifier),
+            checkout_id=checkout_id,
+            patron_id=patron_id,
+            expires=requested_expiry.isoformat(),
+            notification_url=notification_url,
+            passphrase=encoded_pass,
+            hint=self.settings.passphrase_hint,
+            hint_url=self.settings.passphrase_hint_url,
+        )
```

Comment:
This code is just moved around and has not been changed. It was in `get_license_status_document` previously, but only used when that function was called from `_checkout`.